core/src/main/java/hivemall/embedding/CBoWModel.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package hivemall.embedding;
+
+import hivemall.math.random.PRNG;
+import hivemall.utils.collections.maps.Int2FloatOpenHashTable;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+public final class CBoWModel extends AbstractWord2VecModel {
+    protected CBoWModel(final int dim, final int win, final int neg, final int iter,
+            final float startingLR, final long numTrainWords, final Int2FloatOpenHashTable S,
+            final int[] aliasWordId) {
+        super(dim, win, neg, iter, startingLR, numTrainWords, S, aliasWordId);
+    }
+
+    protected void trainOnDoc(@Nonnull final int[] doc) {
+        final int vecDim = dim;
+        final int numNegative = neg;
+        final PRNG _rnd = rnd;
+        final Int2FloatOpenHashTable _S = S;
```

Comment:
Member variable should be `_S` and local variable should be `S`.

`_rnd`, `_aliasWordId` as well.